src/pip/_internal/commands/uninstall.py

```diff
@@ -60,6 +64,12 @@ def run(self, options, args):
             )
             if req.name:
                 reqs_to_uninstall[canonicalize_name(req.name)] = req
+            else:
+                logger.warning(
+                    f"Invalid requirement: '{name}' ignored -"
+                    f" the uninstall command expects named"
+                    f" requirements."
+                )
```

Comment:
```suggestion
                logger.warning(
                    "Invalid requirement: %r ignored -"
                    " the uninstall command expects named"
                    " requirements.",
                    name,
                )
```

[f-string should be avoided in logging in most cases.](https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/)